wlp-managed-8.5/src/main/java/org/jboss/arquillian/container/was/wlp_managed_8_5/WLPManagedContainer.java

```diff
@@ -856,6 +860,37 @@ public void undeploy(Descriptor descriptor) throws DeploymentException {
 
    }
 
+   private void checkForDefinitionExceptions(String applicationName)
+   {
+      String messagesFile = containerConfiguration.getWlpHome() + "/usr/servers/" + containerConfiguration.getServerName() + "/logs/messages.log";
```

Comment:
The log location seems to be defined by the attributes `logDirectory` and `messageFileName` or the respective properties as defined in the knowledge center at https://www.ibm.com/support/knowledgecenter/en/SSEQTP_8.5.5/com.ibm.websphere.wlp.doc/ae/rwlp_logging.html

Do you think it would be easy to retrieve these? Alternatively we could also introduce a new configuration parameter for the wlp-managed-85 container implementation and rely on the developer to provide it.

Reply:
Good suggestion, I will investigate further. I also had another idea to explore if the DefinitionException is nested inside the client observed DeploymentException - and if that might be a better way to detect it. This is why I prefixed the PR as "WIP" as guided by the jboss jira web-page to stand for "Work In Progress" :-) The main reason why I posted the PR now is that I don't want Open Liberty development to be using changed versions of teh container internally without opening them up to upstream. But I agree that this PR should be improved.

Reply:
I think I can use the property values (perhaps set in the bootstrap.properties file) of com.ibm.ws.logging.log.directory and com.ibm.ws.logging.message.file.name or, if these are not set, to use the defaults.

Reply:
I also need to replace the Weld DefinitionException with the javax.enterprise.inject.spi.DefinitionException (its parent)